src/modules/communication/email-service.ts

```diff
@@ -0,0 +1,10 @@
+import SendGridService from './internals/send-grid-service';
+import { SendEmailParams } from './types';
+import { SendGridEmailParams } from './internals/sendgrid-email-params';
+export default class EmailService {
+  public static async sendEmail(params: SendEmailParams): Promise<void> {
+    const sendGridParams = new SendGridEmailParams(params);
```

Comment:
Validation should happen inside `SendGridService.sendEmail`. The root level service should be implementation free